### PR TITLE
Newly created file cannot be opened by other user.

### PR DESCRIPTION
**Pull request by hrishavkmr**

### CODE REVIEW

**Review by manojmalik20 — APPROVED** [review]

File: server/api.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func (p *Plugin) setFilePermissions(fileID, userID, permission string) error {
 		return errors.Wrap(postError, "error when retrieving post for file")
 	}
 
-	if post.UserId != userID {
+	if !skipFileOwnerCheck && post.UserId != userID {
```
Comment (availchet) [inline server/api.go]: Should we have checked for `fileInfo.userID` as in your other PR?

File: server/api.go [hunk 2]
```diff
@@ -127,15 +127,15 @@ func (p *Plugin) handleSaveFilePermissions(w http.ResponseWriter, r *http.Reques
...
+	if err := p.setFilePermissions(fileID, userID, permissionQuery, false); err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
 	returnStatusOK(w)
 }
 
-func (p *Plugin) setFilePermissions(fileID, userID, permission string) error {
+func (p *Plugin) setFilePermissions(fileID, userID, permission string, skipFileOwnerCheck bool) error {
```
Comment (chetanyakan) [inline server/api.go]: nit - rename to `bypassFileOwnerCheck`

File: server/api.go (hunk 1 above)
Comment (hrishavkmr) [reply server/api.go]: actually `fileInfo` doesn't have a userID field. ref - https://pkg.go.dev/github.com/mattermost/mattermost-server/v5/model#FileInfo.
In the web app, the `FileInfo` type did have a `userID` field for the newly created attachment only.

File: server/api.go (hunk 2 above)
Comment (hrishavkmr) [reply server/api.go]: done

**Pull request merged**